src/audio/codec_adapter/codec/cadence.c

```diff
@@ -94,6 +94,34 @@ static struct cadence_api cadence_api_table[] = {
 #endif
 };
 
+static int cadence_codec_resolve_api(struct comp_dev *dev, uint32_t api_id)
+{
+	struct module_data *codec = comp_get_module_data(dev);
+	uint32_t no_of_api = ARRAY_SIZE(cadence_api_table);
+	struct cadence_codec_data *cd = codec->private;
+	int i;
+
+	/* Find and assign API function */
+	for (i = 0; i < no_of_api; i++) {
+		if (cadence_api_table[i].id == api_id) {
+			cd->api = cadence_api_table[i].api;
+			break;
+		}
+	}
+
+	cd->api_id = api_id;
```

Comment:
should this be assigned after the verification below or is this by design - even if the ID isn't found?